src/Components/ProgressBar2.js

```diff
@@ -0,0 +1,94 @@
+import React, { useRef, useEffect } from 'react';
+import { StyleSheet, Animated, View, TouchableOpacity } from 'react-native';
+
+/** Progress bar displayed with an animation which:
+    - shows the number of step;
+    - shows the current step;
+    - enable to move to a given step by clicking on the associated button
+ */
+
+const range = (min, max) => new Array(max - min + 1).fill(0).map((_, i) => min + i);
+
+const Dot = props => {
+  const { idx, full, onPress } = props;
+  const dotStyles = [styles.dot, full ? styles.dotActivated : styles.dotGrey];
+
+  return <TouchableOpacity style={dotStyles} onPress={onPress} key={idx} />;
+};
+
+const useAnimation = targetValue => {
+  const animatedValue = useRef(new Animated.Value(targetValue)).current;
+  const startAnimation = () => {
+    Animated.timing(animatedValue, { toValue: targetValue }).start();
+  };
+  useEffect(startAnimation, [targetValue]);
+  return animatedValue;
+};
+
+export const ProgressBar = props => {
+  const { total, current, onDotPress } = props;
+
+  const width = useAnimation((current - 1) / (total - 1)).interpolate({
+    inputRange: [0, 1],
+    outputRange: ['0%', '100%'],
+  });
+
+  return (
+    <View style={styles.progressContainer}>
+      <View style={styles.progressInner}>
+        <Animated.View style={styles.lineBackground} />
+        <Animated.View style={[styles.lineFiller, { width }]} />
+
+        {range(1, total).map(k => (
```

Comment:
Is there a reason why you use the letter k? Is it a convention?

Reply:
k is for key, but key will be more explicit, will fix.